tests/vxlan/test_vnet_route_leak.py

```diff
@@ -0,0 +1,184 @@
+TESTING_STATUS = "Testing"
+CLEANUP_STATUS = "Cleanup"
+
+SHOW_VNET_ROUTES_CMD = "show vnet routes all"
+SHOW_BGP_SUMMARY_CMD = "show ip bgp summary"
+SHOW_BGP_ADV_ROUTES_CMD_TEMPLATE = "show ip bgp neighbor {} advertised-routes"
+RESTART_BGP_CMD = "sudo systemctl restart bgp"
+CONFIG_SAVE_CMD = "sudo config save -y"
+CONFIG_RELOAD_CMD = "sudo config reload -y"
+BACKUP_CONFIG_DB_CMD = "sudo cp /etc/sonic/config_db.json /etc/sonic/config_db.json.route_leak_orig"
+RESTORE_CONFIG_DB_CMD = "sudo cp /etc/sonic/config_db.json.route_leak_orig /etc/sonic/config_db.json"
+DELETE_BACKUP_CONFIG_DB_CMD = "sudo rm /etc/sonic/config_db.json.route_leak_orig"
+
+BGP_ERROR_TEMPLATE = "BGP sessions not established after {} seconds"
+LEAKED_ROUTES_TEMPLATE = "Leaked routes: {}"
+
+
+@pytest.fixture(scope="module")
+def configure_dut(minigraph_facts, duthost, vnet_config, vnet_test_params):
+    """
+    Setup/teardown fixture for VNET route leak test
+
+    During the setup portion, generates VNET VxLAN configurations and applies them to the DUT
+    During the teardown portion, removes all previously pushed VNET VxLAN information from the DUT
+
+    Args:
+        minigraph_facts: Minigraph information
+        duthost: DUT host object
+        vnet_config: Dictionary containing VNET configuration information
+        vnet_test_params: Dictionary containing VNET test parameters
+    """
+
+    logger.info("Backing up config_db.json")
+    duthost.shell(BACKUP_CONFIG_DB_CMD)
+
+    duthost.shell("sonic-clear fdb all")
+    generate_dut_config_files(duthost, minigraph_facts, vnet_test_params, vnet_config)
+    apply_dut_config_files(duthost, vnet_test_params)
+
+    # In this case yield is used only to separate this fixture into setup and teardown portions
+    yield
+
+    if vnet_test_params[CLEANUP_KEY]:
+        logger.info("Restoring config_db.json")
+        duthost.shell(RESTORE_CONFIG_DB_CMD)
+        duthost.shell(DELETE_BACKUP_CONFIG_DB_CMD)
+
+        duthost.shell(CONFIG_RELOAD_CMD)
+    else:
+        logger.info("Skipping cleanup")
+
+
+def bgp_connected(duthost):
+    """
+    Checks if BGP connections are up
+
+    BGP connections are "up" once they have received all prefixes (6400) from all neighbors
+
+    Args:
+        duthost: DUT host object
+    """
+
+    bgp_summary = duthost.shell(SHOW_BGP_SUMMARY_CMD)["stdout"].split("\n")
+    logger.debug("BGP Summary: {}".format(bgp_summary))
+
+    # Match entire line containing BGP neighbor information such as:
+    # 10.0.0.1        4      64802    6419   10273        0    0    0 00:09:36         6400   ARISTA01T1
+    bgp_neighbor_regex = re.compile(r"^([0-9]{1,3}\.){3}[0-9]{1,3}.*")
+
+    bgp_neighbors = []
+
+    for line in bgp_summary:
+        matched = bgp_neighbor_regex.match(line)
+
+        if matched:
+            bgp_neighbors.append(str(matched.group(0)))
+
+    if not bgp_neighbors:
+        return False
+
+    for neighbor in bgp_neighbors:
+        if neighbor.split()[9] != "6400":
```

Comment:
I don't think we should be looking for `6400 `specific. This may or may not change over time. To make it simple, I would say, you could check if `>0` to confirm it is active. This test is mainly checking for advertised routes and not received routes.

Reply:
Will fix in next push

Reply:
Some dut has a different number. This check will fail. Good catch @prsunny !

Reply:
Do we have a sonichost member function to query bgp and return a structured output (json)? I feel that we could use such a member method to benefit more tests.

Reply:
You can use/improve https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices.py#L1015 and https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices.py#L678

Reply:
AFAIK, we have a json support for this `show ip bgp network json`

Reply:
Now using https://github.com/Azure/sonic-mgmt/blob/d16c7dfa2abc5a25598180043840d78a10ee320f/tests/common/devices.py#L725